### PR TITLE
Create new php-apache-nvm service

### PR DESCRIPTION
**Pull request by agarzola**
## Description

This PR adds a new service called `php-apache-nvm`, to be consumed by projects that aim to manage their Node.js version with nvm.

## Motivation / Context

We would like to declare our Node.js version for a given project using a single canonical declaration: an `.nvmrc` file at the root of the project. This PR creates a whole new Docker image, as opposed to modifying our existing `php-apache` image, because it lowers the stakes. Once we have moved all projects that consume `php-apache` to use `php-apache-nvm` instead, we can simply retire the `php-apache` image altogether.

## Testing Instructions / How This Has Been Tested

An earlier version of this PR modified our `build-images` action to automatically build `php-apache-nvm-test` images and [push them to DockerHub](https://hub.docker.com/r/chromatichq/php-apache-nvm-test/tags). Once I ironed out all the details there, I restored `build-images.yml` and updated the `image-build-php` job to include `php-apache-nvm` as an additional service in both the validation step and the build step. The validation checks in this PR [show that this works as expected](https://github.com/ChromaticHQ/docker-images/actions/runs/1363402755).

In addition to the above, I have [a Draft PR in the CHQ repo](https://github.com/ChromaticHQ/chromatichq.com/pull/3169) which, for the time being, uses the aforementioned `php-apache-nvm-test` image that is still available in DockerHub. As the [Tugboat preview for that PR shows](https://dashboard.tugboat.qa/6152015e7207f2268c2becc4), we are able to use the nvm-installed version of Node.js (and Yarn), albeit with a few modifications to our Tugboat config.

I believe all of the above sufficiently validate that the changes introduced in this PR accomplish our goal without introducing any regressions.

## Screenshots

n/a

## Documentation

n/a — but I’ve left a few PR comments explaining a few things.

### CODE REVIEW

File: .github/workflows/build-images.yml [hunk 1]
```diff
@@ -17,49 +17,27 @@ on:  # yamllint disable-line rule:truthy
...
       - name: Checkout code
         uses: actions/checkout@v2.3.4
 
-      - name: Validate docker images
-        if: ${{ github.event_name == 'pull_request' }}
-        run: docker build services/php-apache/ --build-arg PHP_VERSION=${{ matrix.php }}
-
       - name: Build and push Docker images
         uses: docker/build-push-action@v1
         if: ${{ github.event_name != 'pull_request' }}
```
Comment (agarzola) [inline .github/workflows/build-images.yml:36]: @markdorison I’m opening this Draft PR with the intent to keep it open until we reach a definitive solution. I streamlined this file to only run my temporary `image-build-node-via-nvm` job and I intend to modify this line to build on pushes to this pull request. That way, I can keep this PR open and push to it to trigger new builds as needed to ease experimentation. It will publish new Docker images to a new `chromatichq/php-apache-nvm-test` image in Docker Hub. Here is the change I intend to commit next:
```suggestion
        if: ${{ github.event_name == 'pull_request' }}
```

Before I make that change, I want to make sure I get your eyes on it. The diff makes it a bit hard to follow, so [here’s a handy link to the file in this branch](https://github.com/ChromaticHQ/docker-images/blob/ce263c233949d9d8287260b3ac4d9fc288e1d349/.github/workflows/build-images.yml). Let me know if you have any questions.

Comment (markdorison) [reply .github/workflows/build-images.yml:36]: > Before I make that change, I want to make sure I get your eyes on it.

It's not clear to me what I should be reviewing vs what is just temporary. Could we have a short call to discuss?

Comment (agarzola) [reply .github/workflows/build-images.yml:36]: @markdorison I added this as a topic to our Eng Leads call later this afternoon, since the agenda is a bit thin anyway.

Comment (agarzola) [reply .github/workflows/build-images.yml]: Reviewed during Eng leads call. As discussed, I have fixed the path to the new temporary `Dockerfile` and enabled the new workflow on pull requests.

File: services/php-apache-nvm/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,35 @@
+ARG PHP_VERSION
+FROM tugboatqa/php:${PHP_VERSION}-apache
+ENV COMPOSER_DISCARD_CHANGES=true
+ENV COMPOSER_NO_INTERACTION=true
+ENV COMPOSER_ALLOW_SUPERUSER=true
+RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash - \
+    && . ~/.bashrc \
+    && nvm install --lts \
```
Comment (agarzola) [inline services/php-apache-nvm/Dockerfile:8]: Install nvm, source `.bashrc`, and install LTS version of Node.js.

File: services/php-apache-nvm/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    && echo "#!/bin/bash -l" > /usr/local/bin/install-node-via-nvm.sh \
+    && echo "set -e" >> /usr/local/bin/install-node-via-nvm.sh \
+    && echo "nvm install" >> /usr/local/bin/install-node-via-nvm.sh \
+    && echo "npm install -g yarn" >> /usr/local/bin/install-node-via-nvm.sh \
+    && chmod +x /usr/local/bin/install-node-via-nvm.sh \
```
Comment (agarzola) [inline services/php-apache-nvm/Dockerfile:13]: Create a simple bash script that:
1. Is run with `bash` in login mode (hence the `-l` in the she shebang line).
2. Installs whatever version of Node is specified in `.nvmrc` within the directory that the script is called (or any of its parents).
3. Installs Yarn globally.

You might wonder, “Alfonso, you dum-dum, why not just install Yarn during image build?”, to which I would say: “Because packages installed via npm are specific to the version of Node under which they are installed, so if the consuming project installs a different version of Node.js, then they would not automatically get Yarn. Who’s the dum-dum now??”

File: .github/workflows/build-images.yml [hunk 4]
```diff
@@ -21,24 +21,25 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
+        service: ["php-apache", "php-apache-nvm"]
         php: [7.3, 7.4, "8.0"]
```
Comment (agarzola) [inline .github/workflows/build-images.yml:25]: Adding a dimension to this matrix makes it so that `php-apache` and `php-apache-nvm` images are built for every version of `php` that we specify here.

File: .github/workflows/build-images.yml [hunk 5]
```diff
@@ -21,24 +21,25 @@ jobs:
...
     steps:
       - name: Checkout code
         uses: actions/checkout@v2.3.5
 
       - name: Validate docker images
         if: ${{ github.event_name == 'pull_request' }}
-        run: docker build services/php-apache/ --build-arg PHP_VERSION=${{ matrix.php }}
+        run: docker build services/${{ matrix.service }}/ --build-arg PHP_VERSION=${{ matrix.php }}
```
Comment (agarzola) [inline .github/workflows/build-images.yml:32]: You can see in the checks for this PR that [a `php-apache-nvm` image is created for every `php` version](https://github.com/ChromaticHQ/docker-images/actions/runs/1363402755), in addition to the expected `php-apache` images.

File: .github/workflows/build-images.yml [hunk 6]
```diff
@@ -21,24 +21,25 @@ jobs:
...
         uses: docker/build-push-action@v1
         if: ${{ github.event_name != 'pull_request' }}
         with:
-          path: services/php-apache
+          path: services/${{ matrix.service }}
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_PASSWORD }}
           build_args: PHP_VERSION=${{ matrix.php }}
-          repository: chromatichq/php-apache
+          repository: chromatichq/${{ matrix.service }}
```
Comment (agarzola) [inline .github/workflows/build-images.yml:42]: Since we validated the docker images in this PR, I expect this to build `php-apache-nvm` images to DockerHub, in addition to the expected `php-apache` images.

File: services/php-apache-nvm/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+ENV COMPOSER_ALLOW_SUPERUSER=true
+ENV INSTALL_NODE_SCRIPT_PATH=/usr/local/bin/install-node-yarn-via-nvm
+RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash - \
+    && . ~/.bashrc \
+    && nvm install --lts \
+    && echo "#!/bin/bash -l" > $INSTALL_NODE_SCRIPT_PATH \
+    && echo "set -e" >> $INSTALL_NODE_SCRIPT_PATH \
+    && echo "nvm install" >> $INSTALL_NODE_SCRIPT_PATH \
+    && echo "npm install -g yarn" >> $INSTALL_NODE_SCRIPT_PATH \
+    && chmod +x $INSTALL_NODE_SCRIPT_PATH \
```
[the remark this answers is not in the document]
Comment (adamzimmermann) [reply services/php-apache-nvm/Dockerfile:14]: 🔥

[the remark this answers is not in the document]
Comment (adamzimmermann) [reply services/php-apache-nvm/Dockerfile:14]: Should we add something to the README that documents `/usr/local/bin/install-node-yarn-via-nvm` and what it does/how to use it?

**Review by adamzimmermann — APPROVED** [review]
👏🏼
